### PR TITLE
lib/github: GitHub loader

### PR DESCRIPTION
**Pull request by Morriar**
Helper tool to download a Github repo and store it in a Mongo database.

Features:
* supports github wallets
* save current job and restore jobs after failure
* options to enable branches, commits, issues, pulls, comments and events loading

### CODE REVIEW

File: lib/json/store.nit [hunk 1]
```diff
@@ -111,7 +111,7 @@ class JsonStore
 
 	# Is there data are stored under `key`.
 	fun has_key(key: String): Bool do
-		return (store_dir / "{key}.json").file_exists
+		return ("{store_dir}/{key}.json".simplify_path).file_exists
```
Comment (privat) [inline lib/json/store.nit]: I'm not sure to understand how this fix paths. `simplify_path` does nothing signifiant (before and after paths are equivalent). and using a literal `\` in a string is fragile in border-line cases whereas the `/` operator know how to handle there cases.

File: lib/github/api.nit [hunk 2]
```diff
@@ -621,6 +621,46 @@ class Commit
...
+	var message: nullable String is writable
+end
+
+# Git user authoring data
+class GitUser
+	super GithubEntity
+	serialize
+
+	# Authoring date.
+	var date: nullable String = null is writable
```
Comment (privat) [inline lib/github/api.nit]: what is an authoring date of a user? are users authored?

File: lib/github/loader.nit [hunk 3]
```diff
@@ -0,0 +1,597 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module loader
+
+import config
+import github::wallet
+# import github::cache
```
Comment (privat) [inline lib/github/loader.nit]: is the commented import an easter egg or some kind of psychological warfare?

**Comment by privat** [comment]
test this please

**Comment by privat** [comment]
some unadressed comments. reply to them or label the thing to okwillmerge if you do not care

File: lib/json/store.nit (hunk 1 above)
Comment (Morriar) [inline lib/json/store.nit]: The problem with the `/` operator is that it doesn't like the directory starting with `.` like in `.github_data`.

~~~nit
print ".github_data" / "foo" # prints "/foo"
~~~

So I use the literal `/` to make the path then use `simplify_path` to remove the useless `///` from the path.

File: lib/github/api.nit (hunk 2 above)
Comment (Morriar) [inline lib/github/api.nit]: Yes, when Github doesn't know the account associated to the commiter/author, it create a special kind of user with the signed-off authoring date.

The modelisation is a little sketchy to accommodate the type constraints in the deserializer.

File: lib/github/loader.nit (hunk 3 above)
Comment (Morriar) [inline lib/github/loader.nit]: Removed useless commented import.

File: lib/json/store.nit (hunk 1 above)
Comment (privat) [inline lib/json/store.nit]: I'm not sure I get it.

~~~
$ nit -e 'print ".github_data" / "foo" '
.github_data/foo
~~~

Comment (Morriar) [inline lib/json/store.nit]: $ nit -e 'print ".github_data" / "/foo" '
/foo

**Comment by privat** [comment]
test this please

**Pull request merged**